extensions/eventtap/internal.m

```diff
@@ -189,33 +191,45 @@ static int eventtap_start(lua_State* L) {
 ///  * None
 ///
 /// Returns:
-///  * None
+///  * The event tap object
 static int eventtap_stop(lua_State* L) {
     eventtap_t* e = luaL_checkudata(L, 1, USERDATA_TAG);
 
-    if (!e->running)
-        return 0;
-
-    remove_event(L, e->self);
-    e->self = LUA_NOREF;
-    e->running = false;
-
-    CGEventTapEnable(e->tap, false);
-    CFMachPortInvalidate(e->tap);
-    CFRunLoopRemoveSource(CFRunLoopGetMain(), e->runloopsrc, kCFRunLoopCommonModes);
-    CFRelease(e->runloopsrc);
-    CFRelease(e->tap);
+    if (e->tap && CGEventTapIsEnabled(e->tap)) {
```

Comment:
Is there a situation where we might find that the tap is not enabled, but e->tap still exists? If so, we'd want to clean up some of the resources there too.